doc/Requirements.md

```diff
@@ -0,0 +1,526 @@
+(Why: the parameter structure may be useful for the syncer to communicate with the hardware.)
+
+## Plugins
+The PF MUST be able to create syncers.
+(Why: to bind on the corresponding parameters.)
+
+### Definition
+The PF creates syncer using syncer builder.
+
+### Identifier
+All syncer mapping the same hardware SHOULD have their builder regrouped in a syncer library.
+(Why: to be able to link a group of parameters and a given hardware.)
+
+A syncer builder MUST have a unique identifier in its containing syncer library.
+(Why: To uniquely identify the syncer that should bind on parameters.
+      Given that the syncer library has already been specified.)
+
+A syncer library MUST have a unique identifier in the host system.
+(Why: To identify the library associated to parameters.)
+
+Syncer build and syncer library identifiers SHOULD be strings.
+(Why: The reference application does so.)
+
+### Loading
+Syncer library or/and builder MAY be loaded from dynamically linked libraries (called syncer plugins).
+(Why: The reference implementation supports it.)
+
+Such syncer plugins SHOULD have an unique entry point that -when called- should register
+its payload (syncer library/builder) in the provided gatherer.
+(Note: This permit to merge multiple syncer libraries in one dll.)
+(Why: The reference implementation supports it.)
+
+Multiple syncer plugins, may depend on each other. The PF should appropriatly handle the case
+and not fail.
+(Why: The reference implementation supports it.)
+
+## Mapping
+### Definition
+A parameter not bound to a syncer is called a "virtual parameter".
+(Todo: remove if not used in the requirements.)
+
+### Requirements
+**TODO**:
+ - Plugins
+ - association builder <-> parameters
+
+## Sync
+
+Syncer SHOULD synchronise the mapped hardware on parameter change.
+(Why: To always keep synchronise the underlined hardware and the PF parameters.)
+
+Syncer SHOULD retrieve parameter value from the hardware if no value has be set since the PF start.
+(Note: This is usually implemented on PF start, initialize the parameter values with the mapped hardware current state.)
+(Why: To allow introspection of the hardware.)
+
+A mode with synchronisation on client request MAY be supported.
+(Why: The reference implementation supports it.)
+
+# Rule based dynamic abstraction
+
+## Philosophy
+
+The PF offers parameters mapped on hardware. This is a good but weak abstraction.
+There is often a 1/1 relation between a parameter and the hardware it maps.
+
+A PF offers a mechanism to abstract the parameters to more high level concept.
+
+The goal is to hide numerous parameters and their dynamic value behind simple
+and human friendly api.
+
+It works by regrouping parameters with similar management and defining configurations
+for each "scenario". These "scenario" are then given a priority and a detection predicate.
+Configuration are applied when their associated "scenario" is detected.
```

Comment:
Configuration is applied when its associated "scenario" is detected or
Configurations are applied when their associated "scenarii" are detected